tensorflow/contrib/cmake/tf_shared_lib.cmake

```diff
@@ -76,10 +76,18 @@ add_library(tensorflow SHARED
     ${tensorflow_deffile}
 )
 
+# There is a bug in GCC 5 resulting in undefined reference to a __cpu_model function when
+# linking to the tensorflow library. Adding the following libraries fixes it.
+# See issue on github: https://github.com/tensorflow/tensorflow/issues/9593
+if(CMAKE_COMPILER_IS_GNUCC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.0)
+    set(tensorflow_ADDITIONAL_LIBRARIES gcc_s gcc)
```

Comment:
nit: instead of creating a new tensorflow_ADDITIONAL_LIBRARIES variable, it might be cleaner to add this check below L90 and call target_link_libraries again (according to the cmake docs, it looks like multiple calls append).